src/Raven.Server/Utils/PoolOfThreads.cs

```diff
@@ -40,16 +37,28 @@ public class PoolOfThreads : IDisposable, ILowMemoryHandler
         public PoolOfThreads()
         {
             LowMemoryNotification.Instance?.RegisterLowMemoryHandler(this);
+
+            _cleanupTimer = new Timer(Cleanup, null, TimeSpan.FromMinutes(5), TimeSpan.FromMinutes(5));
```

Comment:
Why go with counter at all?

https://github.com/ravendb/ravendb/blob/ef2f259e1fd53c6d341fd935286008a9c792e8f0/src/Raven.Server/Utils/PoolOfThreads.cs#L268

We can add an `if (_pooled.Count > 64) return; ` there and that would be much simpler, no?

Reply:
I do not think this would work well on a high multi-tenant environment?

Reply:
Why? 64 threads is a _lot_.
Let's consider the scenario of a database with 100 indexes closing. Then reopening.
Is that a scenario that we _want_ to keep threads around?

Reply:
That is a valid scenario in heavy multi-tenant environments. I don't see a reason why not to keep threads around. It's likely that another db will go up and we'll need those 100 threads for indexes. We do have users like that.